pkg/apis/core/validation/validation.go

```diff
@@ -3001,10 +2999,6 @@ func validatePodDNSConfig(dnsConfig *core.PodDNSConfig, dnsPolicy *core.DNSPolic
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("nameservers").Index(i), ns, "must be valid IP address"))
 			}
 		}
-		// Validate searches.
-		if len(dnsConfig.Searches) > MaxDNSSearchPaths {
```

Comment:
I'll defer approval of the runtime implications of this PR to node/network folks.

for the API implications, if we relax validation, it has to be done over two releases so that we don't allow data to be persisted the previous version of the API server considers invalid. 

Release 1: tolerate objects with more search paths on update, but keep the current limit when creating objects or updating objects that meet the current limit

release 2: relax the limit unconditionally

Reply:
Also, do we want to remove the limit completely and make an unbounded list (unlikely), or relax it to be larger?

Reply:
@liggitt 
Thanks for your comment.

> it has to be done over two releases

Did you mean that this patch has to be done over two **_minor_** releases, right? 

> Release 1: tolerate objects with more search paths on update, but keep the current limit when creating objects or updating objects that meet the current limit

I am not sure about this. Are you saying that objects should be created with the current limit but they can be updated beyond the limit?

Reply:
I want to enable it in 1.21.x as an alpha gate. If we decide to proceed as an alpha gate, do we need a KEP process?

Reply:
As Tim said, is too late for 1.21 and you can not merge into a minor release new features https://github.com/kubernetes/kubernetes/pull/100583#issuecomment-808589281 , it has to go in 1.22.

Agree with Tim that a small KEP with a feature gate seems better and cleaner than adding the `isLegacy` argument to the validation functions, also, this is similar to the `kubelete exec probes issue`, and they implemented the KEP + FeatureGate solution  too https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1972-kubelet-exec-probe-timeouts